resources/assets/lib/beatmap-discussions/love-confirmation.tsx

```diff
@@ -0,0 +1,181 @@
+
+    if (this.state.selectedBeatmapIds.has(beatmapId)) {
+      newSelectedIds.delete(beatmapId);
+    } else {
+      newSelectedIds.add(beatmapId);
+    }
+
+    this.setState({ selectedBeatmapIds: newSelectedIds });
+  };
+
+  private handleCheckboxMode = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const mode = e.target.value as GameMode;
+    const modeBeatmapIds = this.getModeBeatmapIds(mode);
+    const newSelectedIds = new Set(this.state.selectedBeatmapIds);
+
+    if (this.checkIsModeSelected(mode) === true) {
+      modeBeatmapIds.forEach((id) => {
+        newSelectedIds.delete(id);
+      });
+    } else {
+      modeBeatmapIds.forEach((id) => {
+        newSelectedIds.add(id);
+      });
+    }
+
+    this.setState({ selectedBeatmapIds: newSelectedIds });
+  };
+
+  private love = () => {
+    if (this.state.selectedBeatmapIds.size === 0) {
+      return;
+    }
+
+    LoadingOverlay.show();
+
+    const url = route('beatmapsets.love', { beatmapset: this.props.beatmapset.id });
+    const params = {
+      data: { beatmapIds: [...this.state.selectedBeatmapIds] },
+      method: 'PUT',
+    };
+
+    $.ajax(url, params).done((response) => {
+      $.publish('beatmapsetDiscussions:update', { beatmapset: response });
+      this.props.onClose();
+    }).fail((xhr) => {
+      osu.ajaxError(xhr);
+    }).always(() => {
+      LoadingOverlay.hide();
+    });
+  };
+
+  private renderDiffMode(mode: GameMode, beatmaps: BeatmapJson[]) {
+    if (beatmaps.length === 0) {
+      return null;
+    }
+
+    return (
+      <div key={mode} className='love-confirmation__diff-mode'>
+        <div className='love-confirmation__diff-mode-title'>
+          <label className='osu-switch-v2'>
+            <input
+              checked={this.checkIsModeSelected(mode) !== false}
+              className='osu-switch-v2__input'
+              data-indeterminate={this.checkIsModeSelected(mode) === 'indeterminate'}
+              onChange={this.handleCheckboxMode}
+              type='checkbox'
+              value={mode}
+            />
+            <span className='osu-switch-v2__content' />
+            <div className='love-confirmation__switch-text'>
+              {osu.trans(`beatmaps.mode.${mode}`)}
+            </div>
+          </label>
+        </div>
+        <ul className='love-confirmation__diff-list'>
+          {beatmaps.map((beatmap) => (
+            <li
+              key={beatmap.id}
+              className='love-confirmation__diff-list-item'
+            >
+              <label className='osu-switch-v2'>
+                <input
+                  checked={this.state.selectedBeatmapIds.has(beatmap.id)}
+                  className='osu-switch-v2__input'
+                  onChange={this.handleCheckboxDifficulty}
+                  type='checkbox'
+                  value={beatmap.id}
+                />
+                <span className='osu-switch-v2__content' />
+                <div className='love-confirmation__switch-text'>
```

Comment:
Main reason is I want to avoid awkward break line like this.

![Screenshot from 2021-07-30 00-10-00](https://user-images.githubusercontent.com/4964985/127535302-6339e338-cbc7-4994-8c78-1e4c906feb1f.png)

Reply:
```
<label> {
  display: flex;
  align-items: baseline;
}
```